[Bug report] Glue catalog logs a misleading warning when Iceberg metadata file is missing

### Version

main branch

### Describe what's wrong

When loading an Iceberg table from the Glue catalog, Gravitino logs a warning saying that partitioning and sort order information may be incomplete.

However, the actual root cause is that the Iceberg metadata JSON file referenced by the Glue table does not exist in S3. The current warning is misleading because it makes the issue look like a partition/sort recovery problem, while the table metadata file itself is missing.

### Error message and/or stacktrace

```text
WARN GlueCatalogOperations.loadTable - Failed to load Iceberg metadata for table <database>.<table>. Partitioning and sort order information may be incomplete.

org.apache.iceberg.exceptions.NotFoundException: Location does not exist:
s3://<bucket>/<table>/metadata/<version>.metadata.json

Caused by: software.amazon.awssdk.services.s3.model.NoSuchKeyException:
The specified key does not exist. (Service: S3, Status Code: 404)
```

### How to reproduce

1. Configure a Gravitino Glue catalog with an Iceberg table.
2. Make the Glue table reference an Iceberg metadata location whose S3 object no longer exists.
3. Load the table through Gravitino, or trigger a path that calls `GlueCatalogOperations.loadTable`.
4. Check the server log.

Expected behavior: the log should clearly indicate that the Iceberg metadata file is missing or cannot be loaded.

Actual behavior: the log says partitioning and sort order information may be incomplete, which hides the real cause.

### Additional context

The warning is emitted from `GlueCatalogOperations.loadTable` after `GlueIcebergTableHelper.loadTable` fails. The catch block handles all exceptions with the same partition/sort warning, so missing metadata files are reported with an inaccurate message.

--- a/catalogs/catalog-glue/src/main/java/org/apache/gravitino/catalog/glue/GlueCatalogOperations.java
+++ b/catalogs/catalog-glue/src/main/java/org/apache/gravitino/catalog/glue/GlueCatalogOperations.java
@@ -377,12 +377,7 @@ public class GlueCatalogOperations implements CatalogOperations, SupportsSchemas
         try {
           GlueIcebergTableHelper.loadTable(icebergGlueCatalog, dbName, ident.name(), table);
         } catch (Exception e) {
-          LOG.warn(
-              "Failed to load Iceberg metadata for table {}.{}. "
-                  + "Partitioning and sort order information may be incomplete.",
-              dbName,
-              ident.name(),
-              e);
+          LOG.warn("Failed to load Iceberg metadata for table {}.{}", dbName, ident.name(), e);
         }
       }
 

--- a/catalogs/catalog-glue/src/test/java/org/apache/gravitino/catalog/glue/TestGlueCatalogOperationsForIceberg.java
+++ b/catalogs/catalog-glue/src/test/java/org/apache/gravitino/catalog/glue/TestGlueCatalogOperationsForIceberg.java
@@ -209,7 +209,7 @@ class TestGlueCatalogOperationsForIceberg {
   }
 
   @Test
-  void testLoadTable_icebergMetadataLoadFailureKeepsGlueOpsContext() {
+  void testLoadTableWithIcebergMetadataLoadFailure() {
     software.amazon.awssdk.services.glue.model.Table rawTable =
         software.amazon.awssdk.services.glue.model.Table.builder()
             .name(TABLE)

--- a/catalogs/catalog-glue/src/test/java/org/apache/gravitino/catalog/glue/TestGlueCatalogOperationsForIceberg.java
+++ b/catalogs/catalog-glue/src/test/java/org/apache/gravitino/catalog/glue/TestGlueCatalogOperationsForIceberg.java
@@ -20,6 +20,7 @@ package org.apache.gravitino.catalog.glue;
 
 import static org.apache.gravitino.catalog.glue.GlueConstants.ICEBERG_TABLE_TYPE_VALUE;
 import static org.apache.gravitino.catalog.glue.GlueConstants.TABLE_TYPE_PARAM;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
@@ -31,6 +32,7 @@ import static org.mockito.Mockito.when;
 import java.util.Map;
 import java.util.Set;
 import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.rel.SupportsPartitions;
 import org.apache.gravitino.rel.TableChange;
 import org.apache.gravitino.rel.expressions.distributions.Distributions;
 import org.apache.gravitino.rel.expressions.transforms.Transform;
@@ -42,8 +44,11 @@ import org.apache.iceberg.catalog.Catalog;
 import org.apache.iceberg.catalog.TableIdentifier;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
 import software.amazon.awssdk.services.glue.GlueClient;
+import software.amazon.awssdk.services.glue.model.GetPartitionsRequest;
+import software.amazon.awssdk.services.glue.model.GetPartitionsResponse;
 import software.amazon.awssdk.services.glue.model.GetTableRequest;
 import software.amazon.awssdk.services.glue.model.GetTableResponse;
 import software.amazon.awssdk.services.glue.model.StorageDescriptor;
@@ -201,6 +206,33 @@ class TestGlueCatalogOperationsForIceberg {
                 Distributions.NONE,
                 null,
                 Indexes.EMPTY_INDEXES));
+  }
+
+  @Test
+  void testLoadTable_icebergMetadataLoadFailureKeepsGlueOpsContext() {
+    software.amazon.awssdk.services.glue.model.Table rawTable =
+        software.amazon.awssdk.services.glue.model.Table.builder()
+            .name(TABLE)
+            .parameters(Map.of(TABLE_TYPE_PARAM, ICEBERG_TABLE_TYPE_VALUE))
+            .storageDescriptor(StorageDescriptor.builder().build())
+            .build();
+
+    when(mockClient.getTable(any(GetTableRequest.class)))
+        .thenReturn(GetTableResponse.builder().table(rawTable).build());
+    when(mockIcebergCatalog.loadTable(any(TableIdentifier.class)))
+        .thenThrow(new RuntimeException("metadata file is missing"));
+    when(mockClient.getPartitions(any(GetPartitionsRequest.class)))
+        .thenReturn(GetPartitionsResponse.builder().build());
+
+    GlueTable result = ops.loadTable(NameIdentifier.of("cat", "ns", DB, TABLE));
+    SupportsPartitions partitions = result.supportPartitions();
+
+    assertEquals(0, partitions.listPartitionNames().length);
+    ArgumentCaptor<GetPartitionsRequest> captor =
+        ArgumentCaptor.forClass(GetPartitionsRequest.class);
+    verify(mockClient).getPartitions(captor.capture());
+    assertEquals(DB, captor.getValue().databaseName());
+    assertEquals(TABLE, captor.getValue().tableName());
   }
 
   @Test

